10.9-libcxx/dpkg.info

```diff
@@ -1,134 +1,235 @@
+perl -pi -e 's,amd64,x86_64,g' data/*table scripts/Makefile.in scripts/Makefile.am scripts/t/Dpkg_Shlibs/basictags.c scripts/t/Dpkg_Shlibs/basictags.symbols scripts/t/Dpkg_Shlibs/objdump.basictags-x86_64 scripts/t/Dpkg_Shlibs.t
+perl -pi -e "s,'-','-' && *p != '_',g" lib/dpkg/arch.c
+perl -pi -e "s,amd64_test,amd64:test,g" lib/dpkg/t/t-arch.c
+
+# add aarch64 for darwin
+/usr/bin/awk '/arm64/{print "arm\t\taarch64\t\t(aarch64|arm64)\t\t64\tlittle"}1' data/cputable > data/cputable.new
+mv data/cputable.new data/cputable
+/usr/bin/awk '/darwin/{print "base-bsd-darwin-arm\t\tdarwin-arm64"}1' data/tupletable > data/tupletable.new
+mv data/tupletable.new data/tupletable
+
+# Fix for tar (Slightly modified for dpkg-bootstrap)
+if [ "%n" = "dpkg-bootstrap" ]; then \
+	if [[ $(uname -a | cut -d' ' -f3) > 12 ]]; then \
+		perl -pi -e 's,\"tar\",\"/usr/bin/tar\",g' dpkg-deb/build.c dpkg-deb/extract.c lib/dpkg/dpkg.h lib/dpkg/subproc.c; \
+		perl -pi -e 's,--warning=no-timestamp,-v,g' dpkg-deb/extract.c; \
+		perl -pi -e 's,--format=gnu,--format=ustar,g' dpkg-deb/build.c; \
+	else \
+		perl -pi -e 's,\"tar\",\"/usr/bin/gnutar\",g' dpkg-deb/build.c dpkg-deb/extract.c lib/dpkg/dpkg.h lib/dpkg/subproc.c; \
+		perl -pi -e 's,--warning=no-timestamp,--format=gnu,g' dpkg-deb/extract.c; \
+	fi \
+else \
+	perl -pi -e 's,\"tar\",\"%p/bin/gnutar\",g' dpkg-deb/build.c dpkg-deb/extract.c lib/dpkg/dpkg.h lib/dpkg/subproc.c; \
+fi
+
+# Fix for older gcc ie: 10.5
+perl -pi -e 's,-Wvla ,,g' configure
+
+# Move dselect and dpkg-dev locals for cleaner Install Phases
+perl -pi -e 's,localedir = \$\(datadir\)\/locale,localedir = \$(datadir)/locale-dselect,g' dselect/Makefile.in
+perl -pi -e 's,localedir = \@localedir\@,localedir = \@localedir\@-dselect,g' dselect/po/Makefile.in.in
+perl -pi -e 's,localedir = \@localedir\@,localedir = \@localedir\@-perl,g' scripts/Makefile.in scripts/po/Makefile.in.in
+
+# Do not require autotools
+perl -pi -e 's,ACLOCAL=.*$,ACLOCAL=echo,g' configure
+perl -pi -e 's,AUTOCONF=.*$,AUTOCONF=echo,g' configure
+perl -pi -e 's,AUTOMAKE=.*$,AUTOMAKE=echo,g' configure
+perl -pi -e 's,AUTOHEADER=.*$,AUTOHEADER=echo,g' configure
+perl -pi -e 's,ACLOCAL_AMFLAGS =.*$,ACLOCAL_AMFLAGS = ,g' Makefile.in
+
+# No librt here
+perl -pi -e 's,-lrt ,,g' lib/dpkg/t/Makefile.*
+
+# Fix Tests
+perl -pi -e 's,dd,gdd,g' t-func/deb-format.at t-func/deb-split.at
+perl -pi -e 's,test_needs_srcdir_switch[\(][\)];,test_needs_author();\ntest_needs_srcdir_switch();,g' t/module-version.t 
+
+# Fix Dpkg.pm values
+perl -pi -e "s,'make','/usr/bin/make',g" scripts/Dpkg.pm
+perl -pi -e "s,'tar','%p/bin/gnutar',g" scripts/Dpkg.pm
+perl -pi -e "s,'/etc/dpkg','%p/etc/dpkg',g" scripts/Dpkg.pm
+perl -pi -e "s,'/var/lib/dpkg','%p/var/lib/dpkg',g" scripts/Dpkg.pm
+
+# Reduce the perl requirement
+perl -pi -e 's,5.20.2,5.16.2,g' configure
+
+# just for building don't force GNU patch (we don't want patch to be essential)
+perl -pi -e "s,GNU patch,patch,g" configure
```

Comment:
```suggestion
perl -pi -e "s,grep -q '.GNU patch',egrep -q '^GNU patch|^patch',g" configure
```
The original change _only_ accepted system patch, and no Fink patch in the path, but the installed dpkg TestDepends on `patch`, so this was breaking rebuild with tests enabled.

Reply:
Good call, I'll work on fixing it to but like your second answer, I don't like using egrep in  PatchScript

`perl -pi -e "s,GNU patch,[\(]GNU [\)]patch,g" configure`

this should work I'm going to run some tests, it'll take a bit though as I"m not setup for testing today.

Reply:
This creates a `grep -q '^[(]GNU [)]?patch'` for me which does not work on either version.
Don't know how to do it without extended re, but
`perl -pi -e "s,(grep -q) '.GNU patch',\${1}E '^(GNU )\?patch',g" configure`
worked in my Monterey-arm64 install.

Reply:
> This creates a `grep -q '^[(]GNU [)]?patch'` for me which does not work on either version.
> Don't know how to do it without extended re, but
> `perl -pi -e "s,(grep -q) '.GNU patch',\${1}E '^(GNU )\?patch',g" configure`
> worked in my Monterey-arm64 install.

cause you didn't wait for me to commit it, I said I hadn't tested yet and to wait.

Reply:
Sorry, thought you'd commit what you had first commented. This one works, both the committed version and the one without `\` in the substitution.